src/Widget/Widget.php

```diff
@@ -53,13 +53,14 @@ public function __construct(EventDispatcherInterface $eventDispatcher, WebView $
      * A matching {@see end()} call should be called later. As some widgets may use output buffering, the {@see end()}
      * call should be made in the same view to avoid breaking the nesting of output buffers.
      *
+     * @param WebView $view
      * @return Widget the newly created widget instance.
      *
      * {@see end()}
      */
-    public static function begin(): Widget
+    public static function begin(WebView $view): Widget
     {
-        $widget = new static(self::$eventDispatcher, self::$webView);
+        $widget = $view->widget(static::class);
 
         self::$stack[] = $widget;
```

Comment:
Widgets stack should be stored at view level. There should be no static properties in `Widget`.